internal/http/listener.go

```diff
@@ -74,7 +78,8 @@ func (listener *httpListener) Accept() (conn net.Conn, err error) {
 	select {
 	case result, ok := <-listener.acceptCh:
 		if ok {
-			return result.conn, result.err
+			return deadlineconn.New(result.conn).
```

Comment:
@klauspost does grid send pings all the time in all situations ?

I was thinking for long listing in a overloaded system, ListDir() does not use grid yet and probably peer-rest-storage can take more than 3 minutes to give a first response. peer-rest-client will most likley be calling Read() and will be stuck there. So we probably need to increase 3 minutes to a higher value.

I can test this theory by adding fake code.

Reply:
you can; I just hoped we had a way to tell if it was an incoming S3 API call or an internode call.

Reply:
I can see grid is complaining without changing anything in the code. 

```
API: SYSTEM()                                                                                  
Time: 14:20:07 UTC 02/09/2024                                                                  
DeploymentID: 9ecbdd07-e7d8-48cc-bcd6-93b32426b606                  
Error: ws read: read tcp 127.0.0.1:45768->127.0.0.1:9002: read: connection reset by peer (*fmt.wrapError)                                                                                      
       2: internal/logger/logger.go:257:logger.LogIfNot()                            
       1: internal/grid/connection.go:962:grid.(*Connection).handleMessages.func1() 
```

Reply:
@vadmeste Connections are monitored and reconnected after ~30s if no traffic flows. This is done by the "write" side on each end. Writes therefore have a timeout.

I can add ListDir to https://github.com/minio/minio/pull/19004 if that is what you are seeing.